### ISSUE TITLE
Opengrok history caching for GIT taking long after moving to a new server.

### ISSUE THREAD

**Issue by trivia21** [oracle/opengrok#2137]
Hi Team,

We are migrating Opengrok version 0.11 to a different server with exact same cpu/memory configuration.
On new server we are facing issue with HistoryCaching of GIT repository, its taking long time to complete which was not the case earlier. I am not sure what specific setting am I missing. Can you help us with that ?

JAVA Option used:
JAVA_OPTS="$JAVA_OPTS -Xmx8192m -Xms2000m"

Below are the only difference which I can see between the servers:
Old Server                                          New Server
CentOS release 6.3  ----------------> Red Hat 6.8
jre-1.6.0-openjdk.x86_64  ---------->jre-1.7.0-openjdk.x86_64

Please let me know if additional details are required.
Regards,
Satish

**Comment by tulinkry** [comment]
try disabling renamed files
https://github.com/oracle/opengrok/blob/master/OpenGrok#L96

**Comment by trivia21** [comment]
I don't see renamed file option in Opengrok 0.11.

**Comment by tulinkry** [comment]
Aha, sorry. I didn't see that info.

**Comment by Orviss** [comment]
0.11 is pretty old. Upgrade is not an option?

**Comment by trivia21** [comment]
Upgrade is in plan but not now and the thing is it was working fine on old server, we just migrated it to a new site on new server and started facing the issue.

**Comment by vladak** [comment]
Couple of questions:
- what history caching do you mean ? Incremental/full
- is this specific to Git repositories only ?
- is this specific to just some Git repositories out of all Git repositories on the system ?
- can you give some reference times from Indexer logs ? Before/after
- what about storage. Has it changed ?

**Comment by trivia21** [comment]
what history caching do you mean ? Incremental/full
=> Incremental

is this specific to Git repositories only ?
=>Yes

is this specific to just some Git repositories out of all Git repositories on the system ?
=>ALL

can you give some reference times from Indexer logs ? Before/after
=>
Before:
18:36:34 INFO: Creating historycache for /opengrok/source/P1 took (7201200ms)
18:36:42 INFO: Creating historycache for /opengrok/source/P2 took (7179471ms)
18:37:09 INFO: Creating historycache for /opengrok/source/P3 took (7235923ms)
19:14:13 INFO: Creating historycache for /opengrok/source/P4 took (7356712ms)
19:16:13 INFO: Creating historycache for /opengrok/source/P5 took (3556942ms)

After:
22:55:38 INFO: Creating historycache for /opengrok/source/P1 took (28164477ms)
22:58:19 INFO: Creating historycache for /opengrok/source/P2 took (28316243ms)
22:47:41 INFO: Creating historycache for /opengrok/source/P3 took (27687386ms)
23:44:47 INFO: Creating historycache for /opengrok/source/P4 took (31040511ms)
22:58:26 INFO: Creating historycache for /opengrok/source/P5 took (28248354ms)

what about storage. Has it changed ?
Storage is same , some mount option are missing will try adding it:
timeo=600,nocto

**Comment by vladak** [comment]
Do you see entries like `Executing command` and `Finished command` in the indexer log ? If yes, it might be useful to compare the time stamps/duration of those `git log` commands as well.

**Comment by tulinkry** [comment]
> No can't see entries like that in log :(

For that you need to decrease your loggin level https://github.com/oracle/opengrok#57-logging to at least FINER (or maybe this is FINEST I'm not sure)

**Comment by vladak** [comment]
w.r.t. patch, it will be possible to disable history per project once #1728 is done, however you will still need to upgrade to the latest to benefit from that.

**Comment by vladak** [comment]
Any new data ?

**Comment by vladak** [comment]
If there is feel free to reopen.
